src/s3/util/fileViewerManager.ts

```diff
@@ -2,30 +2,237 @@
+
+            if (lastModifiedInS3! <= birthtime) {
+                showOutputMessage(`good to retreive, last modified date is before creation`, this.outputChannel) //TODOD: debug log, remove
+                getLogger().debug(`good to retreive, last modified date is before creation`)
+
+                //await this.listTempFolder()
+                return targetLocation
+            } else {
+                getLogger().debug(`last modified date is after creation date!!, removing file and redownloading`)
+                showOutputMessage(
+                    `last modified date is after creation date!!, removing file and redownloading`,
+                    this.outputChannel
+                ) //TODOD: debug log, remove
+
+                fs.unlinkSync(targetPath)
+                return undefined
+            }
+        }
+        return undefined
+    }
+
+    async promptUserConfirm(message: string): Promise<string | undefined> {
+        //for some reason sizeBytes is undefined
+        const cancelButtonLabel = 'Cancel'
+        const confirmButtonLabel = 'Continue with download'
+
+        const result = await this.window.showInformationMessage(message, cancelButtonLabel, confirmButtonLabel)
+        if (result === cancelButtonLabel) {
+            return undefined
+        }
+        return undefined
+    }
+
+    public createTargetPath(fileNode: S3FileNode): Promise<string> {
+        const completePath = getStringHash(readablePath(fileNode)) //TODOD:: map hashes to real name
+        //completePath = completePath.slice(4) //removes 's3://' from path
+
+        //const splittedPath = completePath.split('/')
+        //completePath = splittedPath.join('%')
+
+        return Promise.resolve(path.join(this.tempLocation!, 'S3%' + completePath))
+    }
+
+    async refreshNode(fileNode: S3FileNode): Promise<S3FileNode | undefined> {
+        const parent = fileNode.parent
+        parent.clearChildren()
+
+        await this.commands.execute('aws.refreshAwsExplorerNode', fileNode)
+        await this.commands.execute('aws.refreshAwsExplorerNode', parent)
+        await this.commands.execute('aws.loadMoreChildren', parent)
+
+        const children = await parent.getChildren()
+
+        children.forEach(child => {
+            if ((child as any).name === fileNode.name) fileNode = child as S3FileNode
+        })
+
+        return fileNode
+    }
+
+    /*
+    //TODOD:: remove helper method
+    private listTempFolder(): Promise<void> {
+        getLogger().debug('-------contents in temp:')
+        showOutputMessage('-------contents in temp:', this.outputChannel)
+
+        fs.readdirSync(this.tempLocation).forEach((file: any) => {
+            showOutputMessage(` ${file}`, this.outputChannel)
+            getLogger().debug(` ${file}`)
+        })
+
+        getLogger().debug('-------------------------')
+        showOutputMessage('-------------------------', this.outputChannel)
+        return Promise.resolve()
+    }*/
```

Comment:
Remove fully if you're not going to use it.

Reply:
Will be used it for future work on implementation. I will remove it once the whole feature is ready to be merged.